pkg/controller/direct/mappings/map.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mappings
+
+import "reflect"
+
+type TypeMapping interface {
+	FromType() reflect.Type
+	ToType() reflect.Type
+
+	Map(in *point, out *point) error
+}
+
+type structTypeMapping struct {
+	scope *Mapping
+
+	inType  *reflectType
```

Comment:
Could you add some comments for each field? It's a bit hard to understand how they should be used properly.

Reply:
Good point - done!